### PR TITLE
Wrap arithmetic functions and use numpy names

### PR DESCRIPTION
**Pull request by jl-wynen**
Rename functions
- plus -> add
- minus -> subtract
- times -> multiply

and *_equal variants to match their numpy counterparts.

Also wraps them in Python to provide a better docstring. This especially matters for `mod` do document our particular choice of definition.

### CODE REVIEW

File: python/src/scipp/_arithmetic.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+
+def add(a: DataArrayLike, b: DataArrayLike) -> DataArrayLike:
+    """Element-wise addition.
+
+    Equivalent to
+
+    .. code-block:: python
+
+        a + b
+
```
Comment (OwenArnold) [inline python/src/scipp/_arithmetic.py:20]: Is it worth referring to this https://scipp.github.io/user-guide/computation.html ?

File: python/src/scipp/_arithmetic.py [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+    """
+    return _call_cpp_func(_cpp.add, a, b)
+
+
+def divide(dividend: DataArrayLike, divisor: DataArrayLike) -> DataArrayLike:
+    """Element-wise true division.
+
+    This function corresponds to the ``__truediv__`` dunder method, i.e.
+    ``dividend / divisor``.
+    The result always contains factional parts even when
```
Comment (OwenArnold) [inline python/src/scipp/_arithmetic.py:33]: typo fractional

File: python/src/scipp/_arithmetic.py [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+
+    .. code-block:: python
+
+        minuend - subtrahend
+
+    :param minuend: Minuend.
+    :param subtrahend: Subtrahend.
+    :return: ``subtrahend`` subtracted from ``minuend``.
+    """
+    return _call_cpp_func(_cpp.subtract, minuend, subtrahend)
```
Comment (OwenArnold) [inline python/src/scipp/_arithmetic.py:158]: Overall, documentation in wrappers is nice!

**Review by OwenArnold — CHANGES_REQUESTED** [review]
One minor typo spotted. Also, you don't mention this in release notes even though potentially a breaking change. I'm not aware of anyone doing `sc.times(a, b)` but it is possible they were.

File: python/src/scipp/_arithmetic.py (hunk 1 above)
Comment (jl-wynen) [reply python/src/scipp/_arithmetic.py:20]: Good idea

File: docs/about/release-notes.rst [hunk 4]
```diff
@@ -17,6 +17,8 @@ Features
 Breaking changes
 ~~~~~~~~~~~~~~~~
 
+* Changes names of arithmetic functions to match numpy's names: ``add``, ``subtract``, ``multiply`` `#1999 <https://github.com/scipp/scipp/pull/1999>`_.
```
Comment (OwenArnold) [inline docs/about/release-notes.rst:20]: Minor point, maybe list the exact names deprecated, for easy look up by users, can be done at some later point in any case.

**Review by OwenArnold — APPROVED** [review]

**Review by OwenArnold — APPROVED** [review]

**Pull request merged**